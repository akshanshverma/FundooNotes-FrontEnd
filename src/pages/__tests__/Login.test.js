import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from '@material-ui/core/Button';

import Login from "../Login";

const component = shallow(<Login debug />);
describe('Login', () => {
    it('should render correctly in "debug" mode', () => {
        expect(component).toMatchSnapshot();
    });

    // it('renders a email input', () => {
    //     expect(component.find('#userEmail').length).toEqual(1);
    // })

    // it('renders a password input', () => {
    //     expect(component.find('#userPassword').length).toEqual(1);
    // })

    // it('should have register redirect option', () => {
    //     expect(component.find('span.reg').length).toEqual(1);
    // })

    // it('should have forget password redirect option', () => {
    //     expect(component.find('span.psw').length).toEqual(1);
    // })

    it('should change event and email set in state ', () => {
        const emailInput = component.find('#userEmail');
        emailInput.props().onChange({
            target: {
                name: 'email',
                value: 'akshansh.verma01@gamil.com'
            }
        });
        expect(component.state('email')).toEqual('akshansh.verma01@gamil.com');
    });


    it('should change event and password set in state ', () => {
        const emailInput = component.find('#userPassword');
        emailInput.props().onChange({
            target: {
                name: 'password',
                value: 'qwerty'
            }
        });
        expect(component.state('password')).toEqual('qwerty');
    });
});

describe('test button component which is use for login',()=>{
    it('should click for user login',()=>{
        const mockFun = jest.fn();
        const button = shallow(<Button children='ch' onClick={mockFun}/>);
        button.simulate('click');
        expect(mockFun.mock.calls.length).toEqual(1);
    });
});
