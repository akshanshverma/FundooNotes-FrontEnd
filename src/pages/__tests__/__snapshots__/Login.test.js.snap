// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render correctly in "debug" mode 1`] = `
<div
  className="maindiv"
>
  <WithStyles(Card)
    id="card"
  >
    <WithStyles(Typography)
      color="primary"
      id="loginT"
    >
      Login
    </WithStyles(Typography)>
    <div
      className="hold"
    >
      <div
        className="input"
      >
        <Input
          id="userEmail"
          label="email"
          name="email"
          onChange={[Function]}
          placeholder="enter email"
          type="text"
        />
      </div>
      <div
        className="input"
      >
        <Input
          id="userPassword"
          label="password"
          name="password"
          onChange={[Function]}
          placeholder="enter password"
          type="password"
        />
      </div>
      <div
        className="button"
      >
        <WithStyles(Button)
          color="primary"
          id="loginButton"
          onClick={[Function]}
          type="Submit"
          variant="contained"
        >
          login
        </WithStyles(Button)>
      </div>
      <div
        className="errMsg"
      />
      <div
        className="socialLogin"
      >
        <LoginWithGoogle
          loginWithGoogle={[Function]}
        />
        <LoginWithFacebook
          loginWithGoogle={[Function]}
        />
      </div>
      <div
        id="spanDiv"
      >
        <span
          className="reg"
        >
          New User? 
          <a
            href="/register"
          >
            register
          </a>
        </span>
        <span
          className="psw"
        >
          Forgot 
          <a
            href="/forgetpassword"
          >
            password?
          </a>
        </span>
      </div>
    </div>
  </WithStyles(Card)>
</div>
`;
